synapse/storage/databases/main/events.py

```diff
@@ -1845,6 +1845,18 @@ def _handle_chunk_event(self, txn: LoggingTransaction, event: EventBase):
             },
         )
 
+        # When we receive an event with a `chunk_id` referencing the
+        # `next_chunk_id` of the insertion event, we can remove it from the
+        # `insertion_event_extremities` table.
+        sql = """
+            DELETE FROM insertion_event_extremities WHERE event_id IN (
+                SELECT event_id FROM insertion_events
+                WHERE next_chunk_id = ?
+            )
+        """
```

Comment:
I'm unsure if this query is correct 🎯

---

```
# Doesn't seem to be supported in postgres
sql = """
    DELETE extrem FROM insertion_event_extremities as extrem
    INNER JOIN insertion_events AS i USING (event_id)
    WHERE i.next_chunk_id = ?
"""
```

```
sql = """
    DELETE FROM insertion_event_extremities WHERE event_id IN (
        SELECT event_id FROM insertion_events
        WHERE next_chunk_id = ?
    )
"""
```

Reply:
I think the query is fine. With postgres you can use a [`USING`](https://www.postgresql.org/docs/10/sql-delete.html) clause to join against tables, but that isn't supported in SQLite.